File: practica1/P1S3/FilterChain.cpp
#pragma once
#include "Filtro.cpp"
#include "Filtro1.cpp"
#include "Filtro2.cpp"
#include <vector>
using namespace std;

class FilterChain
{
    private:
        vector<Filtro*> *filtros;
    
    public:
        FilterChain()
        {
            filtros = new vector<Filtro*>();
        }

        void aniadirFiltro(Filtro *f)
        {
            (*filtros).push_back(f);
        }

        void ejecutarFiltros()
        {
            for(int i = 0; i < filtros->size(); i++)
            {
                (*filtros)[i]->ejecutar();
            }
                
        }

        void vaciarFiltro()
        {
            (*filtros).clear();
        }
};

File: practica1/P1S3/Java/src/practica1dss3/FilterChain.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practica1dss3;

import java.util.ArrayList;

/**
 *
 * @author USUARIO
 */
public class FilterChain {
    private ArrayList<Filtro> filtros;
    
    public FilterChain(){
        filtros = new ArrayList<>();
    }
    
    public void añadirFiltro(Filtro f){
        filtros.add(f);
    }
    
    public void ejecutarFiltros(){
        for(int i=0; i<filtros.size(); i++)
            filtros.get(i).ejecutar();
    }
   
    public void vaciarFiltros(){
        filtros.clear();
    }
}
